traffic_incident_parser/include/traffic_incident_parser_worker.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2019-2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#ifndef TRAFFIC_INCIDENT_PARSER_WORKER_H
+#define TRAFFIC_INCIDENT_PARSER_WORKER_H
+
+#include <ros/ros.h>
+#include <cav_msgs/MobilityOperation.h>
+#include <cav_msgs/TrafficControlMessageV01.h>
+#include <functional>
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <string>
+#include <cstring>
+#include <vector>
+#include <algorithm>
+#include <sstream>
+#include <lanelet2_extension/projection/local_frame_projector.h>
+#include <lanelet2_core/geometry/Lanelet.h>
+#include <lanelet2_routing/RoutingGraph.h>
+#include <carma_wm/WorldModel.h>
+#include <std_msgs/String.h>
+#include <gps_common/GPSFix.h>
+#include <carma_wm/Geometry.h>
+
+namespace traffic{
+
+class TrafficIncidentParserWorker
+{
+
+ public:
+
+  using PublishTrafficControlCallback = std::function<void(const cav_msgs::TrafficControlMessageV01&)>;
+
+  /*!
+   * \brief Constructor
+   */
+  TrafficIncidentParserWorker(carma_wm::WorldModelConstPtr wm,PublishTrafficControlCallback traffic_control_pub);
+    
+  /*! \fn projectionCallback(const std_msgs::String &projection_msg)
+    \brief projectionCallback to convert ECEF to MAP frame
+    \param  std_msgs::String
+  */
+
+  void projectionCallback(const std_msgs::String &projection_msg);
+
+    /*! \fn mobilityOperationCallback(const cav_msgs::MobilityOperation &mobility_msg)
+    \brief mobilityOperationCallback to receive the incoming mobility operation message from message node
```

Comment:
also add that this function publishes the geofence message upon processing the mobility msg